lib/handlers.py

```diff
@@ -306,18 +310,44 @@ def _request_completions_old(cls, view, data):
             cls._last_location = data['cursor_runes']
         cls._run_auto_complete(view)
 
-    @staticmethod
-    def _run_auto_complete(view):
+    @classmethod
+    def _run_auto_complete(cls, view):
         # It seems like the `auto_complete` command does not always result in
         # `on_query_completions` from being triggered if a completion list is
         # currently shown, so we need to hide it first.
-        view.run_command('hide_auto_complete')
+        # we're only hiding if we're not displaying a subset of the previous set
+        # Sublime is handling the subset correctly
+        if not cls._is_completions_subset():
+            view.run_command('hide_auto_complete')
+
         view.run_command('auto_complete', {
             'api_completions_only': True,
             'disable_auto_insert': True,
             'next_completion_if_showing': False,
         })
 
+    @classmethod
+    def _is_completions_subset(cls):
+        with cls._lock:
+            # both sets of completions are in the Kite's original data format
+            previous = cls._visible_completions
+            current = cls._received_completions
+
+        if len(previous) == 0 or (len(current) > len(previous)):
+            return False
+
+        index_previous = 0
```

Comment:
You should be able to rewrite this as:

```python
for item, index in enumerate(current):
  if item != previous[index]:
    return False
```

`enumerate` is a more Pythonic way to access both the item and index when iterating over a list. Also, since we established that `len(current) <= len(previous)` in line 336 above, we know that `previous[index]` will never be out of bounds.